src/dmd/typesem.d

```diff
@@ -1651,13 +1651,21 @@ extern(C++) Type typeSemantic(Type t, Loc loc, Scope* sc)
             case TOK.variable:
                 mtype.sym = (cast(VarExp)e).var;
                 break;
+            case TOK.function_:
+                if ((cast(FuncExp)e).td)
```

Comment:
Isn’t there a method to avoid the casting?

Reply:
Can't you use `e.isFuncExp`?

Reply:
Not sure what's best.

Reply:
Ah, Thank you. I wasn't sure this.

Reply:
Hmm, `isFuncExp` uses `cast(FuncExp)` internally and checks if `e.op == TOK.function_` one more. I would like to go without change.

Reply:
Ok.